main.cpp

```diff
@@ -44,28 +44,376 @@ Purpose:  This project continues developing Project3.
+            break;
+        }
+    }
+}
+
+void Fridge::crushIce(unsigned int ice)
+{
+    std::cout << ice << "Ice cubes crushed! " << '\n';
+}
+
+double Fridge::streamWater(double tankVolume)
+{
+    return (tankVolume - 1.0);
+}
+
+unsigned int Fridge::adjustTemperature(Door door)
+{
+    bool switchedOn = door.switchOnDisplay(true);
+
+    return switchedOn ? true : false;
+}
+
+bool Fridge::openDoor(Door door, int angle = 45)
+{
+    for(auto i = 0; i <= angle; ++i)
+    {
+        door.angle = i;
+        std::cout << "Door angle " << i << "degrees!" << '\n';
+    }
+
+    return true;
+}
+
 /*
  copied UDT 3:
  */
 
+struct Amplifier 
+{ 
+    Amplifier();
+    ~Amplifier();
+
+    unsigned int inputs;
+    unsigned int outputs {4};
+    char inputType {'B'};
+    unsigned int wattage;
+    double maxVolume ;
+
+    struct Preamp
+    {
+        Preamp();
+
+        unsigned int inputResistance {1000};
+        unsigned int outputResistance {8};
+        float amplificationFactor {10.0f};
+        float thdNoise;
+        float voltage;
+
+        void increaseAmplification(float ampFactor);
+        void switchOutResistance(unsigned int outRes);
+        void switchInResistance(unsigned int inRes);
+    };
+
+    Preamp preA;
+
+    void adjustVolume(Preamp pre);
+    bool switchOn(); 
+    bool changeInp(char inputType); 
+    void ejectCd(int time);
+}; 
+
+Amplifier::Amplifier() : inputs(8), wattage(45), maxVolume(10.0)
+{
+    std::cout << "Amplifier created!" << '\n';
+}
+
+Amplifier::~Amplifier()
+{
+    std::cout << "Amplifier Object destruceted!" << '\n';
+}
+
+Amplifier::Preamp::Preamp() : thdNoise(0.0001f), voltage(12)
+{
+    std::cout << "Preamp created!" << '\n';
+}
+
+void Amplifier::Preamp::increaseAmplification(float ampFac)
+{
+    std::cout << "Factor set: " << ampFac << '\n';
+}
+
+void Amplifier::Preamp::switchOutResistance(unsigned int outR)
+{
+    std::cout << "Output Resistance set: " << outR<< '\n';
+}
+
+void Amplifier::Preamp::switchInResistance(unsigned int inR)
+{
+    std::cout << "Input Resistance set: " << inR<< '\n';
+}
+
+void Amplifier::adjustVolume(Preamp pre)
+{
+    pre.increaseAmplification(5);
+}
+
+bool Amplifier::switchOn()
+{
+    return true;
+}
+
+bool Amplifier::changeInp(char type)
+{
+    switch(type) 
+    {
+        case 'A' :
+            std::cout << "Class A" << '\n'; 
+            break;
+        case 'B' :
+            std::cout << "Class B" << '\n';
+            break;
+        default :
+            std::cout << "Invalid type" << '\n';
+   }
+
+    return (type == ('A' | 'B') ? true : false);
+}
+
+void Amplifier::ejectCd(int time)
+{
+    auto i {0};
+    while(i != time)
+    {
+        ++i;
+    }
+    std::cout << "Ejecting CD took " << i << " seconds" << '\n';
+}
+
 /*
  new UDT 4:
  with 2 member functions
  */
 
+struct Kitchen
+{
+    Kitchen();
+    ~Kitchen();
+
+    Fridge fridge;
+    CoffeeMachine machine;
+    bool table {false}; // true if something is on the table
```

Comment:
This is a terrible variable name. 
`bool tableIsOccupied` is a good variable name.

Boolean variables should be named around the idea that they store if something IS or IS NOT [doing/storing/ other -ing words]

- [ ] revise